giraffe/src/components/GeoLayer.tsx

```diff
@@ -92,13 +92,22 @@ const updateQuery = (
       lat,
       zoom
     )
-    onUpdateQuery(variableAssignment)
-    lastRenderProperties.latOnLastRender = lat
-    lastRenderProperties.lonOnLastRender = lon
-    lastRenderProperties.zoomOnLastRender = zoom
-    lastRenderProperties.widthOnLastRender = width
-    lastRenderProperties.heightOnLastRender = height
+    if (delayedQueryTimerHandle) {
+      clearInterval(lastRenderProperties.delayedQueryTimerHandle)
+    }
+    // we don't want to call query update too often (on window resize)
+    // we get decoupled from onUpdateQuery call errors
+    lastRenderProperties.delayedQueryTimerHandle = setTimeout(() => {
```

Comment:
Suggestion: Can we move this into `utils`? So we can clean this component to render only what it's purpose is.

Reply:
I don't understant what exactly you would like to move out of this component. Do you mean the setTimeout ? The code you are reffering to is preventing multiple calls to influxdb as the display window is resized. It is an additional condition to prevent querying influxdb with the same parameters repeatedly. I think that my previous comment applies here as well.